src/Data/RRBVector/Internal/Debug.hs

```diff
@@ -85,3 +88,113 @@ pattern Leaf :: Array a -> Tree a
 pattern Leaf arr <- RRB.Leaf arr
 
 {-# COMPLETE Balanced, Unbalanced, Leaf #-}
+
+-- | Structural invariants a vector is expected to hold.
+data Invariant
+    = RootSizeGt0      -- Root: Size > 0
+    | RootShiftDiv     -- Root: The shift at the root is divisible by blockShift
+    | RootSizeCorrect  -- Root: The size at the root is correct
+    | RootGt1Child     -- Root: The root has more than 1 child if not a Leaf
+    | BalShiftGt0      -- Balanced: Shift > 0
+    | BalNumChildren   -- Balanced: The number of children is blockSize unless
+                       -- the parent is unbalanced or the node is on the right
+                       -- edge in which case it is in [1,blockSize]
```

Comment:
I think the number of children should still be `blockSize` if the parent is unbalanced (except when the node is on the right edge ofc).

Reply:
That doesn't seem to be the case. For example,
```hs
λ> showTree (0 <| fromList [1..256])
"Root {size = 257, shift = 8, tree = Unbalanced [Balanced [Leaf [0]],Balanced [Leaf [1,2,3,...,15,16], <...14 more leaves...> ,Leaf [241,242,...,255,256]]] ([1,257])}"
```
The newly consed leaf is balanced but has only one child.

Reply:
You're right. I was thinking about something like
```hs
Root {size = ..., shift = 8, tree = Unbalanced [Balanced [Leaf [0],Leaf[1]], Balanced [...] ([2,...])}
```
but that's already covered by `BalNumChildren`.